### PR TITLE
Add nvtx-c

### PR DESCRIPTION
**Pull request by shwina**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/libnvtx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  name: libnvtx
+  version: {{ version }}
+
+source:
+  url: https://github.com/NVIDIA/NVTX/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: mkdir -p "$PREFIX"/include && cp -r c/include/nvtx3 "$PREFIX"/include  # [not win]
```
Comment (scopatz) [inline recipes/libnvtx/meta.yaml:14]: This should be in a `build.sh`

Comment (shwina) [reply recipes/libnvtx/meta.yaml]: Thanks! Fixed.

Apologies, but this PR is not ready to merge yet as we are still giving some thought to the package name. I converted this PR to a draft.

**Comment by jcohen-nvidia** [comment]
This change looks good to me.  I like that the nesting of nvToolsExt.h underneath the nvtx3 directory is preserved, so code can do #include <nvtx3/nvToolsExt.h> to ensure it is picking up the exact right version.  The CUDA toolkit for example provides version 2 of NVTX in the same include directory as cuda.h, so this practice of using the nvtx3 directory in the #includes is the safest way to avoid accidentally picking up old headers.

**Comment by shwina** [comment]
Changed the package name from `libnvtx` to `nvtx-c`. This should now be ready for review @scopatz. Apologies again for the confusion!

**Comment by shwina** [comment]
@conda-forge-admin, please ping team

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**